observatory-dags/observatory/dags/telescopes/crossref_events.py

```diff
@@ -0,0 +1,388 @@
+        :param dataset_id: the dataset id.
+        :param merge_partition_field: the BigQuery field used to match partitions for a merge
+        :param updated_date_field: the BigQuery field used to determine newest entry for a merge
+        :param bq_merge_days: how often partitions should be merged (every x days)
+        :param airflow_vars: list of airflow variable keys, for each variable it is checked if it exists in airflow
+        """
+
+        if airflow_vars is None:
+            airflow_vars = [AirflowVars.DATA_PATH, AirflowVars.PROJECT_ID, AirflowVars.DATA_LOCATION,
+                            AirflowVars.DOWNLOAD_BUCKET, AirflowVars.TRANSFORM_BUCKET]
+        super().__init__(dag_id, start_date, schedule_interval, dataset_id, merge_partition_field,
+                         updated_date_field, bq_merge_days, dataset_description=dataset_description,
+                         airflow_vars=airflow_vars)
+
+        self.add_setup_task_chain([self.check_dependencies,
+                                   self.get_release_info])
+        self.add_task_chain([self.download,
+                             self.upload_downloaded,
+                             self.transform,
+                             self.upload_transformed,
+                             self.bq_load_partition])
+        self.add_task_chain([self.bq_delete_old,
+                             self.bq_append_new,
+                             self.cleanup], trigger_rule='none_failed')
+
+    def make_release(self, **kwargs) -> CrossrefEventsRelease:
+        # Make Release instance
+        ti: TaskInstance = kwargs['ti']
+        start_date, end_date, first_release = ti.xcom_pull(key=CrossrefEventsTelescope.RELEASE_INFO,
+                                                           include_prior_dates=True)
+
+        release = CrossrefEventsRelease(self.dag_id, start_date, end_date, first_release)
+        return release
+
+    def download(self, release: CrossrefEventsRelease, **kwargs):
+        """ Task to download the CrossrefEventsRelease release.
+
+        :param release: a CrossrefEventsRelease instance.
+        :return: None.
+        """
+        # Download release
+        release.download()
+
+    def upload_downloaded(self, release: CrossrefEventsRelease, **kwargs):
+        """ Task to upload the downloadeded CrossrefEventsRelease release.
+
+        :param release: a CrossrefEventsRelease instance.
+        :return: None.
+        """
+        # Upload each downloaded release
+        upload_files_from_list(release.download_files, release.download_bucket)
+
+    def transform(self, release: CrossrefEventsRelease, **kwargs):
+        """ Task to transform the CrossrefEventsRelease release.
+
+        :param release: an CrossrefEventsRelease instance.
+        :return: None.
+        """
+        # Transform each release
+        release.transform()
+
+
+def extract_events(url: str, events_path: str, next_cursor: str = None, success: bool = True) -> \
```

Comment:
A name such as `download_events` might convey what this function does a little more.